### ISSUE TITLE
Allow a field with custom schema to be nullable during deserialization

### ISSUE THREAD

**Issue by blaise-io** [paylogic/halogen#16]
Given these schema

```python
class ChildSchema(halogen.Schema):
    some_other_field = halogen.Attr(halogen.types.String())

class ParentSchema(halogen.Schema):
    some_field = halogen.Attr(ChildSchema)
```

To make `some_field` nullable during deserialization, I have to create a `deserialize` class method on `ChildSchema`:

```python
class ChildSchema(halogen.Schema):
    some_other_field = halogen.Attr(halogen.types.String())

    @classmethod
    def deserialize(self, value, **kwargs):
        return None if value is None else super().deserialize(value)
```

It would be nice if `halogen.Attr` supported a `nullable` parameter:  
`some_field = halogen.Attr(ChildSchema, nullable=True)`

Or alternatively support wrapping:  
`some_field = halogen.Attr(halogen.Nullable(ChildSchema))`

**Comment by olegpidsadnyi** [comment]
As an alternative nullable can indicate that default should be used or not. This way it will be similar to required, which indicates if the attr/key has to be present and nullable would indicate if the default should be used if the value is None.

**Comment by olegpidsadnyi** [comment]
Fixed in #17
